packages/docs-app/src/examples/core-examples/numericInputBasicExample.tsx

```diff
@@ -64,154 +64,134 @@ const BUTTON_POSITIONS = [
+    const [buttonPosition, setButtonPosition] = React.useState<NumericInputProps["buttonPosition"]>("right");
+    const [disabled, setDisabled] = React.useState(false);
+    const [fill, setFill] = React.useState(false);
+    const [intent, setIntent] = React.useState<Intent>(Intent.NONE);
+    const [large, setLarge] = React.useState(false);
+    const [leftElement, setLeftElement] = React.useState(false);
+    const [leftIcon, setLeftIcon] = React.useState(false);
+    const [locale, setLocale] = React.useState<string>();
+    const [max, setMax] = React.useState(100);
+    const [min, setMin] = React.useState(0);
+    const [selectAllOnFocus, setSelectAllOnFocus] = React.useState(false);
+    const [selectAllOnIncrement, setSelectAllOnIncrement] = React.useState(false);
+    const [small, setSmall] = React.useState(false);
+    const [value, setValue] = React.useState("");
+
+    const handleInputValueChange = React.useCallback(
+        (_valueAsNumber: number, valueAsString: string) => setValue(valueAsString),
+        [],
     );
 
-    private handleLocaleChange = handleStringChange(locale =>
-        this.setState({ locale: locale === "default" ? undefined : locale }),
-    );
-
-    private toggleDisabled = handleBooleanChange(disabled => this.setState({ disabled }));
-
-    private toggleLeftIcon = handleBooleanChange(leftIcon =>
-        this.setState({ leftIcon: leftIcon ? "dollar" : undefined }),
-    );
+    const handleLargeChange = handleBooleanChange(newLarge => {
+        setLarge(newLarge);
+        if (newLarge) {
+            setSmall(false);
+        }
+    });
 
-    private handleSmallChange = handleBooleanChange(small => this.setState({ small, ...(small && { large: false }) }));
-
-    private handleLargeChange = handleBooleanChange(large => this.setState({ large, ...(large && { small: false }) }));
-
-    private toggleLeftElement = handleBooleanChange(leftElement =>
-        this.setState({
-            leftElement: leftElement ? (
-                <Popover
-                    position="bottom"
-                    content={
-                        <Menu>
-                            <MenuItem icon={IconNames.Equals} text={"Equals"} />
-                            <MenuItem icon={IconNames.LessThan} text={"Less than"} />
-                            <MenuItem icon={IconNames.GreaterThan} text={"Greater than"} />
-                        </Menu>
-                    }
-                >
-                    <Button minimal={true} icon={IconNames.Filter} />
-                </Popover>
-            ) : undefined,
-        }),
+    const handleLocaleChange = handleStringChange(newLocale =>
+        setLocale(newLocale === "default" ? undefined : newLocale),
     );
 
-    private toggleFullWidth = handleBooleanChange(fill => this.setState({ fill }));
+    const handleSmallChange = handleBooleanChange(newSmall => {
+        setSmall(newSmall);
+        if (newSmall) {
+            setLarge(false);
+        }
+    });
```

Comment:
Mega nit: Would prefer this callback is adjacent to `handleLargeChange`

Reply:
bfd8fe651fe565910b60e3cc650f82067a06ceae